include/emul.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_EMUL_H_
+#define ZEPHYR_INCLUDE_EMUL_H_
+
+/**
+ * @brief Emulators used to test drivers and higher-level code that uses them
+ * @defgroup io_emulators Emulator interface
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+struct device;
+struct emul;
+
+/**
+ * Structure uniquely identifying a device that is a child of an i2c_emul node.
+ *
+ * Currently this uses the device node label, but that will go away by 2.5.
+ */
+struct emul_link {
+	const char *label;
+};
+
+/** List of emulators attached to a bus */
```

Comment:
Why are I2C-specific emulator structures here?  I don't think we'll want to add to this file every time a subsystem is extended with emulator support.

`emul_link` was designed taking only I2C into account.  It's *probably* suitable for other things that are bus children, though, so these could be `emul_bus_child_link` and `emul_bus_child_list`,  if you want to generalize off the one instance.

Reply:
I've renamed it to refer to a bus.